flow/src/org/labkey/flow/controllers/FlowController.java

```diff
@@ -310,10 +317,13 @@ public ModelAndView getView(FlowAdminForm form, boolean reshow, BindException er
         @Override
         public boolean handlePost(FlowAdminForm form, BindException errors)
         {
-            if (form.getWorkingDirectory() != null)
+            PipeRoot root = PipelineService.get().findPipelineRoot(getContainer());
```

Comment:
You can probably hang on to and reuse the pipeline root from the `validateCommand` method.